### PR TITLE
Add support for /proc/self(pid)/maps

### PR DESCRIPTION
**Pull request by jessehui**

### CODE REVIEW

File: src/libos/src/process/elf_file.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@ use std::ffi::CStr;
 use std::fmt;
 use std::str;
 
+use crate::fs::{AsINodeFile, INodeFile};
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/process/elf_file.rs]: Done. Thanks for your review.

File: src/libos/src/process/elf_file.rs [hunk 2]
```diff
@@ -107,8 +108,12 @@ impl<'a> ElfFile<'a> {
         self.elf_buf
     }
 
-    pub fn file_inode(&self) -> &Arc<dyn INode> {
-        self.file_inode
+    pub fn file_ref(&self) -> &FileRef {
+        self.file_ref
+    }
+
+    pub fn as_inode_file(&self) -> Result<&INodeFile> {
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/process/elf_file.rs]: Done.

File: src/libos/src/vm/process_vm.rs [hunk 3]
```diff
@@ -244,7 +248,7 @@ impl<'a, 'b> ProcessVMBuilder<'a, 'b> {
                 }
 
                 // Bytes of file_size length are loaded from the ELF file
-                elf_file.file_inode().read_at(
+                elf_file.as_inode_file().unwrap().read_at(
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/vm/process_vm.rs]: Done.

File: src/libos/src/process/do_spawn/exec_loader.rs [hunk 4]
```diff
@@ -105,12 +105,12 @@ pub fn load_file_hdr_to_vec(
         .read_elf64_lazy_as_vec()
         .map_err(|e| errno!(e.errno(), "failed to read the file"))?;
 
-    let elf_header = ElfFile::parse_elf_hdr(&inode, &mut file_buf);
+    let elf_header = ElfFile::parse_elf_hdr(inode, &mut file_buf);
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/process/do_spawn/exec_loader.rs]: Done.

File: src/libos/src/vm/chunk.rs [hunk 5]
```diff
@@ -175,10 +174,10 @@ impl Chunk {
             unreachable!();
         };
         trace!("get lock, try mmap in chunk: {:?}", self);
-        if internal_manager.chunk_manager().free_size() < options.size() {
+        if internal_manager.chunk_manager_mut().free_size() < options.size() {
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/vm/chunk.rs]: Done.

**Review by volcano0dr — APPROVED** [review]
LGTM

**Pull request merged**